src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -286,6 +286,11 @@ static int get_xclbin_mem_topology(const char *buffer, XmaXclbinInfo *xclbin_inf
         const char *data = &buffer[ip_hdr->m_sectionOffset];
         const mem_topology *mem_topo = reinterpret_cast<const mem_topology *>(data);
         auto& xma_mem_topology = xclbin_info->mem_topology;
+        const axlf_section_header *mem_grp_hdr = xclbin::get_axlf_section(xclbin, ASK_GROUP_TOPOLOGY);
+        xclbin_info->has_mem_groups = false;
+	if (mem_grp_hdr) {
```

Comment:
This will always be true.  The key is you don't really know if the xclbin is grouping banks or not.  Not unless you iterate all mem_data entries and search for the bank 'tag' that is used for the groups; and doing this is not really a good idea.

The xclbin produced in 2020.2 is composed by xclbinutil and will always have a GROUP_TOPOLOGY section, but this section may not have groups and could be just a copy of MEM_TOPOLOGY.

Reply:
ok. Better would be if xlcbinutil doesn't make a group_topology which is just a copy of mem_topology. That copy doesn't serve any new purpose. I will make a change in this.

Reply:
Modified